### DISCUSSION TITLE
Need some help with custom highlighting

### DISCUSSION THREAD

**Discussion by imcleo** [icsharpcode/AvalonEdit#311] [Q&A] [answered]
Hi, I'm self-taught. I have a custom scripting language (very simple) that I'm trying to do syntax highlighting for. I've read the docs + searched around the internet but I'm not understanding how to add a .xshd file for the custom highlighting... Could anyone help me please? ^^

**Comment by siegfriedpammer** [comment, marked as answer]
https://github.com/icsharpcode/AvalonEdit/blob/774b38e672b3f4252f9f4e6f2f3f6707ea7fb943/ICSharpCode.AvalonEdit.Sample/Window1.xaml.cs#L44-L55

Hope this helps!

**Reply by imcleo** [reply]
Thanks for the reply! Could you explain to me what the `string[] extensions` param is for RegisterHighlighting?

**Reply by siegfriedpammer** [reply]
It is used by the `HighlightingManager.GetDefinitionByExtension` method. It can be used to find the matching highlighting definition based on a given filename. Note, however, this is not used by AvalonEdit internally, but can be used by library users to retrieve the correct highlighting for a file.
